lib/Lemplate.pm

```diff
@@ -538,18 +538,35 @@ local function stash_set(stash, k, v, default)
     end
 end
 
+_M.filters = {
+    html = function (s, args)
+        s = gsub(s, "&", '&amp;', "jo")
+        s = gsub(s, "<", '&lt;', "jo");
+        s = gsub(s, ">", '&gt;', "jo");
+        s = gsub(s, '"', '&quot;', "jo"); -- " end quote for emacs
+        return s
+    end,
+
+    lower = function (s, args)
+        return string.lower(s)
+    end,
+
+    upper = function (s, args)
+        return string.upper(s)
+    end,
+}
+
 function _M.process(file, params)
     local stash = params
     local context = {
         stash = stash,
         filter = function (bits, name, params)
             local s = concat(bits)
-            if name == "html" then
-                s = gsub(s, "&", '&amp;', "jo")
-                s = gsub(s, "<", '&lt;', "jo");
-                s = gsub(s, ">", '&gt;', "jo");
-                s = gsub(s, '"', '&quot;', "jo"); -- " end quote for emacs
-                return s
+            local f = _M.filters[name]
+            if f then
+                return f(s, params)
+            else
```

Comment:
We should not use `else` when the previous branch never falls through :) I think we encountered this before ;)